codegen/client.go

```diff
@@ -40,19 +40,44 @@ type clientConfig interface {
 	NewClientSpec(
 		instance *ModuleInstance,
 		h *PackageHelper) (*ClientSpec, error)
-	customImportPath() string
-	fixture() *Fixture
 }
 
 // ClientSubConfig is the "config" field in the client-config.yaml for http
 // client and tchannel client.
 type ClientSubConfig struct {
-	ExposedMethods   map[string]string `yaml:"exposedMethods" json:"exposedMethods" validate:"exposedMethods"`
-	CustomImportPath string            `yaml:"customImportPath" json:"customImportPath"`
-	ThriftFile       string            `yaml:"thriftFile" json:"thriftFile" validate:"nonzero"`
-	ThriftFileSha    string            `yaml:"thriftFileSha" json:"thriftFileSha" validate:"nonzero"`
-	SidecarRouter    string            `yaml:"sidecarRouter" json:"sidecarRouter"`
-	Fixture          *Fixture          `yaml:"fixture" json:"fixture"`
+	ExposedMethods map[string]string `yaml:"exposedMethods" json:"exposedMethods" validate:"exposedMethods"`
+	ThriftFile     string            `yaml:"thriftFile" json:"thriftFile" validate:"nonzero"`
+	ThriftFileSha  string            `yaml:"thriftFileSha" json:"thriftFileSha" validate:"nonzero"`
+	SidecarRouter  string            `yaml:"sidecarRouter" json:"sidecarRouter"`
+	TestConfig     `yaml:",inline" json:",inline"`
+}
+
+// TestConfig contains fixtures for generating mock client
+type TestConfig struct {
+	Fixture          *Fixture `yaml:"fixture" json:"fixture"`
+	CustomImportPath string   `yaml:"customImportPath" json:"customImportPath"`
```

Comment:
`CustomImportPath` being a field on `TestConfig` doesn't seem to make sense, it is irrelevant to test